apps/s_server.c

```diff
@@ -586,6 +592,67 @@ static int get_ocsp_resp_from_responder(SSL *s, tlsextstatusctx *srctx,
     return ret;
 }
 
+static int get_ocsp_resp_from_responder(SSL *s, tlsextstatusctx *srctx,
+                                        STACK_OF(OCSP_RESPONSE) **sk_resp)
+{
+    const SSL_CONNECTION *sc = SSL_CONNECTION_FROM_CONST_SSL(s);
+    X509 *x = NULL;
+    int ret = SSL_TLSEXT_ERR_NOACK;
+    int i, num = 1;
+    STACK_OF(X509) *server_certs = NULL;
+    OCSP_RESPONSE *resp = NULL;
+
+    if (*sk_resp != NULL)
+        sk_OCSP_RESPONSE_pop_free(*sk_resp, OCSP_RESPONSE_free);
+
+    *sk_resp = sk_OCSP_RESPONSE_new_null();
+
+    SSL_get0_chain_certs(s, &server_certs);
+    /*
+     * TODO: in future DTLS should also be considered
+     */
+    if (server_certs != NULL && sc != NULL &&
+        !SSL_CONNECTION_IS_DTLS(sc) && SSL_version(s) >= TLS1_3_VERSION) {
+        /* certificate chain is available */
+        num = sk_X509_num(server_certs) + 1;
+    } else {
```

Comment:
As @vdukhovni had pointed out, there are cases (at least with TLS <= 1.2) where there is no server cert, so AFAICS your code needs to set `num` to 0 in case `sc == NULL`.